src/main/java/core/file/hrf/HRFStringBuilder.java

```diff
@@ -79,8 +88,39 @@ public void createArena(Map<String, String> arenaDataMap) {
         appendKeyValue(arenaStringBuilder, "expandingVIP", arenaDataMap.get("ExVIP"));
         appendKeyValue(arenaStringBuilder, "expandingSseatTotal", arenaDataMap.get("ExTotal"));
         appendKeyValue(arenaStringBuilder, "isExpanding", arenaDataMap.get("isExpanding"));
-        // Achtung bei keiner Erweiterung = 0!
-        appendKeyValue(arenaStringBuilder, "ExpansionDate", arenaDataMap.get("ExpansionDate"));
+        appendKeyValue(arenaStringBuilder, "RebuiltDate", Optional.ofNullable(arenaDataMap.get("RebuiltDate")).orElse(""));
+        appendKeyValue(arenaStringBuilder, "ExpansionDate", Optional.ofNullable(arenaDataMap.get("ExpansionDate")).orElse(""));
+    }
+
+    /**
+     * Create the arena data.
+     */
+    public void createArena(Arena arena) {
+        arenaStringBuilder = new StringBuilder("[arena]\n");
+        appendKeyValue(arenaStringBuilder, "arenaname", arena.getName());
+        appendKeyValue(arenaStringBuilder, "arenaid", arena.getId());
+        appendKeyValue(arenaStringBuilder, "antalStaplats", arena.getCurrentCapacity().getTerraces());
+        appendKeyValue(arenaStringBuilder, "antalSitt", arena.getCurrentCapacity().getBasic());
+        appendKeyValue(arenaStringBuilder, "antalTak", arena.getCurrentCapacity().getRoof());
+        appendKeyValue(arenaStringBuilder, "antalVIP", arena.getCurrentCapacity().getVip());
+        appendKeyValue(arenaStringBuilder, "seatTotal", arena.getCurrentCapacity().getTotal());
+        appendKeyValue(arenaStringBuilder, "expandingStaplats", arena.getExpandedCapacity().map(Capacity::getTerraces).orElse(0));
+        appendKeyValue(arenaStringBuilder, "expandingSitt", arena.getExpandedCapacity().map(Capacity::getBasic).orElse(0));
+        appendKeyValue(arenaStringBuilder, "expandingTak", arena.getExpandedCapacity().map(Capacity::getRoof).orElse(0));
+        appendKeyValue(arenaStringBuilder, "expandingVIP", arena.getExpandedCapacity().map(Capacity::getVip).orElse(0));
+        appendKeyValue(arenaStringBuilder, "expandingSseatTotal", arena.getExpandedCapacity().map(Capacity::getTotal).orElse(0));
+        appendKeyValue(arenaStringBuilder, "isExpanding", arena.getExpandedCapacity().isPresent());
+        appendKeyValue(arenaStringBuilder, "RebuiltDate", arena.getCurrentCapacity().getRebuiltDate().map(HRFStringBuilder::localDateTimeToHrfString).orElse(""));
+        appendKeyValue(arenaStringBuilder, "ExpansionDate", arena.getExpandedCapacity().map(ExpandedCapacity::getExpansionDate).map(HRFStringBuilder::localDateTimeToHrfString).orElse(""));
+    }
+
+    // TODO: duplicates from HODateTime
```

Comment:
Yup :-)

Reply:
I have currently no idea to fit that in a nice way but it is on my list.
It bothers me. :-)